knack/output.py

```diff
@@ -142,6 +164,8 @@ def out(self, obj, formatter=None, out_file=None):  # pylint: disable=no-self-us
                   file=out_file, end='')
 
     def get_formatter(self, format_type):  # pylint: disable=no-self-use
+        if not sys.stdout.isatty() and format_type in ['jsonc', 'yamlc']:
+            format_type = format_type[:-1]
```

Comment:
>format_type = format_type[:-1] [](start = 12, length = 30)

it works when you using string slice method to convert jsonc to json, but the code seems not easy to read, and if there is new type need convert like cjson->json, this code is not reusable. Can you explicitly map jsonc to json in your logic? Besides, I notice you add jsonc,ymalc in the OutputProducer._FORMAT_DICT above, what do you return different formatter here?

Reply:
Thank you for the suggestions. Refactored as requested. Also added test for removing color.